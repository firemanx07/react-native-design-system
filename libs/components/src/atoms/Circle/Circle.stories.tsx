import { ColorType } from '@proxym/themes';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import { StorybookScreen } from '../../storybook';
import { Circle as CircleBase } from '../Circle';

const Story: ComponentMeta<typeof CircleBase> = {
  component: CircleBase,
  title: 'Atoms/Circle',
  argTypes: {
    color: {
      control: { type: 'color' },
      options: ColorType,
    },
  },
};
export default Story;
const Template: ComponentStory<typeof CircleBase> = ({ size, color }) => {
  return (
    <StorybookScreen title="Circle :">
      <CircleBase size={size} color={color} />
    </StorybookScreen>
  );
};

const parameters = {
  controls: {
    include: ['size', 'color'],
  },
};

const defaultArgs = {
  size: 20,
  color: '#4D37D9',
};

export const Circle = Template.bind({});
Circle.parameters = parameters;
Circle.args = {
  ...defaultArgs,
};
