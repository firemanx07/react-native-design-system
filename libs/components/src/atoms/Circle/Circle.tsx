import { styled } from '@proxym/themes';
import React, { memo } from 'react';

import { TestIDType } from '../../types';

type CircleProps = {
  size: number;
  color: string;
} & TestIDType;
const Circle = (props: CircleProps) => <Container {...props} />;

const Container = styled.View<CircleProps>`
  background-color: ${({ color }) => color};
  border-radius: ${({ size }) => size / 2};
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;
`;
export default memo(Circle);
