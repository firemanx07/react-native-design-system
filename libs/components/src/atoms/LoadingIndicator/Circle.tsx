import { useNamespacedTheme } from '@proxym/themes';
import React, { useEffect, useCallback, memo } from 'react';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
  interpolate,
} from 'react-native-reanimated';

import { usePrevious } from '../../hooks';

type PropsType = {
  size: number;
  active: boolean;
  betweenSpace: number;
  animationDuration: number;
  scaleTo: number;
  color?: string;
};

const Circle = ({
  active,
  size,
  scaleTo,
  animationDuration,
  betweenSpace,
  color,
}: PropsType) => {
  const prevActive = usePrevious<boolean>(active);
  const theme = useNamespacedTheme();
  const animatedValue = useSharedValue(0);

  const scaleUp = useCallback(() => {
    animatedValue.value = withTiming(1, {
      duration: animationDuration,
    });
  }, [animatedValue, animationDuration]);

  const scaleDown = useCallback(() => {
    animatedValue.value = withTiming(0, {
      duration: animationDuration,
    });
  }, [animatedValue, animationDuration]);

  useEffect(() => {
    if (prevActive && !active) {
      scaleDown();
    }
    if (!prevActive && active) {
      scaleUp();
    }
  }, [prevActive, active, scaleUp, scaleDown]);

  const activeColor = color || theme.colors.dark;

  const animatedStyle = useAnimatedStyle(() => {
    return {
      transform: [
        {
          scale: interpolate(animatedValue.value, [0, 1], [0, scaleTo]),
        },
      ],
      opacity: interpolate(animatedValue.value, [0, 1], [0.5, 1]),
    };
  });

  const style = {
    height: size,
    width: size,
    borderRadius: size / 2,
    marginHorizontal: betweenSpace,
    backgroundColor: activeColor,
  };

  return <Animated.View style={[style, animatedStyle]} />;
};

export default memo(Circle);
