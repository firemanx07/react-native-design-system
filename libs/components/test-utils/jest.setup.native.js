import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
