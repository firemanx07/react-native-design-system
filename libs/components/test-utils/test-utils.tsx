import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
