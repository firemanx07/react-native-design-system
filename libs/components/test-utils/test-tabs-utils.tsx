import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
