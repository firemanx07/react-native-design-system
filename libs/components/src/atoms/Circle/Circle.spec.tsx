import { testAllStories } from '../../../test-utils';
import * as stories from './Circle.stories';

describe('Checkbox tests', () => {
  testAllStories(stories);
});
