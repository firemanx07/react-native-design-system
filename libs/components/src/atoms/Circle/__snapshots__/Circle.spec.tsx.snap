// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox tests Circle snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Circle :
  </Text>
  <View
    color="#4D37D9"
    size={20}
    style={
      [
        {
          "backgroundColor": "#4D37D9",
          "borderRadius": 10,
          "height": 20,
          "width": 20,
        },
      ]
    }
  />
</View>
`;
